internal/pkg/config/config.go

```diff
@@ -262,56 +295,78 @@ func GenerateCrawlConfig() error {
 	}
 
 	if config.MaxContentLengthMiB > 0 {
-		slog.Info("Max content length is set, payload over X MiB would be discarded", "X", config.MaxContentLengthMiB)
+		slog.Info("max content length is set, payload over X MiB would be discarded", "X", config.MaxContentLengthMiB)
 	}
 
 	if config.MaxOutlinks > 0 {
-		slog.Info("Max outlinks is set, only the first X outlinks will be processed", "X", config.MaxOutlinks)
+		slog.Info("max outlinks is set, only the first X outlinks will be processed", "X", config.MaxOutlinks)
 	}
 
 	if config.RandomLocalIP {
-		slog.Warn("Random local IP is enabled")
+		slog.Warn("random local IP is enabled")
 	}
 
 	if config.DisableIPv4 && config.DisableIPv6 {
-		slog.Error("Both IPv4 and IPv6 are disabled, at least one of them must be enabled.")
+		slog.Error("both IPv4 and IPv6 are disabled, at least one of them must be enabled.")
 		os.Exit(1)
 	} else if config.DisableIPv4 {
 		slog.Info("IPv4 is disabled")
 	} else if config.DisableIPv6 {
 		slog.Info("IPv6 is disabled")
 	}
 
+	config.exclusionRegexes.Store([]*regexp.Regexp(nil))
 	if len(config.ExclusionFile) > 0 {
-		for _, file := range config.ExclusionFile {
-			var (
-				regexes []string
-				err     error
-			)
-
-			if strings.HasPrefix(file, "http://") || strings.HasPrefix(file, "https://") {
-				slog.Info("Reading (remote) exclusion file", "file", file)
-				regexes, err = readRemoteExclusionFile(file)
-				if err != nil {
-					return err
+		if config.ExclusionFileLiveReload {
+			config.waitGroup.Add(1)
+			go func() {
+				defer config.waitGroup.Done()
+
+				ticker := time.NewTicker(config.ExclusionFileLiveReloadInterval)
+				defer ticker.Stop()
+
+				for {
+					select {
+					case <-config.ctx.Done():
+						slog.Info("exclusion file live reload goroutine cancelled")
+						return
+					case <-ticker.C:
```

Comment:
Between Zeno boot up and the first exclusion reload trigger, we don't have any exclusions available in the window.

Should load once before starting the live reloader. (and the live loader can be a function in exclusions.go)

```golang
	if len(config.ExclusionFile) > 0 {

		// ...load exclusion...

		config.setExclusionRegexes(exclusions)

		if config.ExclusionFileLiveReload {
			config.waitGroup.Add(1)
			go exclusionFileLiveReloader()
		}
	}
```

Reply:
_Did you mean `config.waitGroup.Go(func() { config.exclusionFileLiveReloader() })` ? :p_ 
Thanks!